api/src/test/java/org/pmiops/workbench/profile/ProfileServiceTest.java

```diff
@@ -77,4 +92,203 @@ public void testReturnsLastAcknowledgedTermsOfService() {
     assertThat(profile.getLatestTermsOfServiceVersion()).isEqualTo(1);
     assertThat(profile.getLatestTermsOfServiceTime()).isEqualTo(1);
   }
+
+  @Test
+  public void validateInstitutionalAffiliation() {
+    VerifiedInstitutionalAffiliation affiliation =
+        new VerifiedInstitutionalAffiliation()
+            .institutionShortName("Broad")
+            .institutionDisplayName("The Broad Institute")
+            .institutionalRoleEnum(InstitutionalRole.OTHER)
+            .institutionalRoleOtherText("Kibitzing");
+
+    Profile profile =
+        new Profile()
+            .verifiedInstitutionalAffiliation(affiliation)
+            .contactEmail("kibitz@broadinstitute.org");
+
+    DbInstitution dbInstitution = new DbInstitution();
+    dbInstitution.setShortName("Broad");
+    dbInstitution.setDisplayName("The Broad Institute");
+
+    when(mockInstitutionDao.findOneByShortName("Broad")).thenReturn(Optional.of(dbInstitution));
+
+    when(mockInstitutionService.validateAffiliation(
+            any(DbVerifiedInstitutionalAffiliation.class), anyString()))
```

Comment:
[opt] I feel like there could be some meaningful argument capture here, to ensure that this method is called with the correct institutional affiliation shortName. This would require building a non-empty DbVerifiedInstitutionalAffiliation at L122, which feels like a good idea anyway (since in the real world, we'll expect that Db model object to be non-empty).

This connection, between the mapper return value and the institutionService call, is IMO the main piece of logic that's being tested in this positive case – so it seems potentially useful to explicitly check.

Reply:
Something like:

```
create mock DbVerifiedInstitutionalAffiliation with appropriate values
tell mapper to return mock when called
tell institution service to return true when called with mock (or when called with anything?)
spy on institutionService validate call to make sure the DbVerifiedInstitutionalAffiliation has the right shortname
```

Wouldn't that be redundant? Did you maybe mean the profileService call?